lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -79,6 +79,13 @@ static bool allStoredPropertiesConformToProtocol(DeclContext *DC,
       return false;
 
     auto type = propertyDecl->getType()->mapTypeOutOfContext();
```

Comment:
You can replace lines 81-88 with `auto type = propertyDecl->getValueInterfaceType()`